firebase_admin/_user_mgt.py

```diff
@@ -372,6 +374,81 @@ def photo_url(self):
     def provider_id(self):
         return self._data.get('providerId')
 
+class ActionCodeSettings(object):
+    """Contains required continue/state URL with optional Android and iOS settings.
+    Used when invoking the email action link generation APIs.
+    """
+    KEYS = set(['url', 'handle_code_in_app', 'dynamic_link_domain', 'ios_bundle_id',
+                'android_package_name', 'android_minimum_version', 'android_install_app'])
+
+    def __init__(self, data=None):
+        super(ActionCodeSettings, self).__init__()
+        data = data or {}
+        if not isinstance(data, dict):
+            raise ValueError('Invalid data argument: {0}. Must be a dictionary.'.format(data))
+        if len(set(six.iterkeys(data)) - self.KEYS):
+            raise ValueError('Invalid settings provided: {0}. Valid dictionary \
+                have following keys {1}'.format(data, self.KEYS))
+        self._data = copy.deepcopy(data)
+
+    @property
+    def url(self):
+        return self._data.get('url', None)
+
+    @url.setter
+    def url(self, url):
+        self._data['url'] = url
+
+    @property
+    def handle_code_in_app(self):
+        return self._data.get('handle_code_in_app', False)
+
+    @handle_code_in_app.setter
+    def handle_code_in_app(self, handle_code_in_app):
+        self._data['handle_code_in_app'] = handle_code_in_app
+
+    @property
+    def dynamic_link_domain(self):
+        return self._data.get('dynamic_link_domain', None)
+
+    @dynamic_link_domain.setter
+    def dynamic_link_domain(self, dynamic_link_domain):
+        self._data['dynamic_link_domain'] = dynamic_link_domain
+
+    @property
+    def ios_bundle_id(self):
+        return self._data.get('ios_bundle_id', None)
+
+    @ios_bundle_id.setter
+    def ios_bundle_id(self, ios_bundle_id):
+        self._data['ios_bundle_id'] = ios_bundle_id
+
+    @property
+    def android_package_name(self):
+        return self._data.get('android_package_name', None)
+
+    @android_package_name.setter
+    def android_package_name(self, android_package_name):
+        self._data['android_package_name'] = android_package_name
+
+    @property
+    def android_minimum_version(self):
+        return self._data.get('android_minimum_version', None)
+
+    @android_minimum_version.setter
+    def android_minimum_version(self, android_minimum_version):
+        self._data['android_minimum_version'] = android_minimum_version
+
+    @property
+    def android_install_app(self):
+        return self._data.get('android_install_app', False)
+
+    @android_install_app.setter
+    def android_install_app(self, android_install_app):
+        self._data['android_install_app'] = android_install_app
+
+    def to_parameters_dict(self):
```

Comment:
Lets remove this, and instead add an `encode_action_code_settings()` method to the module. It should accept an `ActionCodeSettings` instance, and return a `dict`.